### PR TITLE
Error Handling and Two Complete Hash Maps

### PR DESCRIPTION
**Pull request by newdevon**
Error Handling working for all current command line inputs. This handles things like string inputs when a integer is expected. With that we have two hash maps right now. One hash map has all the items and respective pricing. The other hash map has all the users and divided money owed to the recipient.

### CODE REVIEW

File: splitCheck.py [hunk 1]
```diff
@@ -10,62 +10,108 @@
...
 
-    
+def namesInput(user_total):
+    names_list = []
+    while True:
+        try:
+            name_input = input("Enter a payer's name | Type 'DONE' to proceed: ")
+            if name_input == "DONE":
+                if len(names_list) == 0:
+                    raise MinimumValueError
```
Comment (erjiale) [inline splitCheck.py:31]: keyword 'raise' will make you exit the application? If so, maybe better if we just log the error msg, and have them reenter name, etc... Same for the other 'raise'

**Review by erjiale — APPROVED** [review]
Nice job!!!

File: splitCheck.py (hunk 1 above)
Comment (erjiale) [reply splitCheck.py:31]: But doesn't matter too much, I like the exception raises 👍

Comment (xuejingao) [reply splitCheck.py:31]: I agree with the first point, normally when something fails, it doesn't crash the program -- but rather inform the user and let them re-try. 

I think removing the `break` keyword solves this...?

File: splitCheck.py [hunk 2]
```diff
@@ -10,62 +10,108 @@
...
-        item_name = input("What is the item? | Type 'DONE' to proceed: ")
-        if item_name == "DONE":
-            break
-        else: #need to write cases for invalid inputs like non-intger values
+        try:
+            item_name = input("What is the item? | Type 'DONE' to proceed: ")
+            if item_name == "DONE": 
+                if len(items_dict) == 0: 
+                    raise MinimumValueError
+                break
```
Comment (xuejingao) [inline splitCheck.py:56]: Same as above

File: splitCheck.py [hunk 3]
```diff
@@ -10,62 +10,108 @@
...
+                who_list = names
+            else:
+                for char in name_string:
+                    if char.isdigit():
+                        index = int(char)
+                        who_list.append(names[index - 1])
+                number_of_people = len(who_list)
+                if number_of_people == 0:
+                    raise UnwantedStringError
+            break
```
Comment (xuejingao) [inline splitCheck.py:92]: Same as above.

File: splitCheck.py (hunk 1 above)
Comment (newdevon) [reply splitCheck.py:31]: Lol I didn't see this at all. Raise doesn't crash the program. It does what you think should be done it raises the error and then in one of the exceptions I catch it and prompt them to re-enter until a correct input is given.

**Pull request merged**